op-e2e/op_geth.go

```diff
@@ -73,12 +79,32 @@ func NewOpGeth(t *testing.T, ctx context.Context, cfg *SystemConfig) (*OpGeth, e
 		SystemConfig: e2eutils.SystemConfigFromDeployConfig(cfg.DeployConfig),
 	}
 
-	node, _, err := geth.InitL2("l2", big.NewInt(int64(cfg.DeployConfig.L2ChainID)), l2Genesis, cfg.JWTFilePath)
-	require.Nil(t, err)
-	require.Nil(t, node.Start())
+	var (
+		node         Node
+		wssEndpoint  string
+		HTTPEndpoint string
+	)
+	if cfg.ExternalL2Shim == "" {
+		gethNode, _, err := geth.InitL2("l2", big.NewInt(int64(cfg.DeployConfig.L2ChainID)), l2Genesis, cfg.JWTFilePath)
+		require.Nil(t, err)
+		require.Nil(t, gethNode.Start())
+		node = gethNode
```

Comment:
Nit: If we simply created the `EthInstance`, then we could eliminate the pre-declared endpoints, and consolidate lines 84/85/92/93/102/103 into a single pair of lines.  No functional difference though.